lib/lib-storage/src/Upload.ts

```diff
@@ -207,17 +208,17 @@ export class Upload extends EventEmitter {
 
   private async __doConcurrentUpload(dataFeeder: AsyncGenerator<RawDataPart, void, undefined>): Promise<void> {
     for await (const dataPart of dataFeeder) {
-      if (this.uploadedParts.length > this.MAX_PARTS) {
+      if (this.uploadEnqueuedPartsCount > this.MAX_PARTS) {
```

Comment:
reason for this change: `this.uploadedParts.length` is updated after an async operation (UploadPart), which causes a delay. 

Due to request concurrency, `this.uploadedParts.length` can be far behind the actual number of parts that have been iterated and submitted.